core/src/test/scala/unit/kafka/server/KafkaServerTest.scala

```diff
@@ -154,6 +155,96 @@ class KafkaServerTest extends QuorumTestHarness {
+      "org.apache.kafka.server.log.remote.storage.NoOpRemoteLogMetadataManager")
+    tsEnabledProps.put(RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP,
+      "org.apache.kafka.server.log.remote.storage.NoOpRemoteStorageManager")
+
+    val server = TestUtils.createServer(KafkaConfig.fromProps(tsEnabledProps))
+    server.remoteLogManagerOpt match {
+      case Some(_) =>
+      case None => fail("RemoteLogManager should be initialized")
+    }
+
+    val topicProps = new Properties()
+    topicProps.setProperty(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, true.toString)
+
+    TestUtils.createTopic(zkClient = server.zkClient, topic = "batman", servers = Seq(server), topicConfig = topicProps)
+
+    server.shutdown()
+
+    val tsDisabledProps = TestUtils.createBrokerConfigs(1, zkConnect).head
+
+    assertThrows(classOf[ConfigException], () => TestUtils.createServer(KafkaConfig.fromProps(tsDisabledProps)))
+  }
+
+  @Test
+  def testClusterWideDisablementOfTieredStorageWithDisabledTieredTopic(): Unit = {
+    val tsEnabledProps = TestUtils.createBrokerConfigs(1, zkConnect).head
+    tsEnabledProps.put(RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP, true.toString)
+    tsEnabledProps.put(RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP,
+      "org.apache.kafka.server.log.remote.storage.NoOpRemoteLogMetadataManager")
+    tsEnabledProps.put(RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP,
+      "org.apache.kafka.server.log.remote.storage.NoOpRemoteStorageManager")
+
+    var server = TestUtils.createServer(KafkaConfig.fromProps(tsEnabledProps))
+    server.remoteLogManagerOpt match {
+      case Some(_) =>
+      case None => fail("RemoteLogManager should be initialized")
+    }
+
+    val topicProps = new Properties()
+    topicProps.setProperty(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, false.toString)
+
+    TestUtils.createTopic(zkClient = server.zkClient, topic = "batman", servers = Seq(server), topicConfig = topicProps)
+
+    server.shutdown()
+
+    val tsDisabledProps = TestUtils.createBrokerConfigs(1, zkConnect).head
+
+    server = TestUtils.createServer(KafkaConfig.fromProps(tsDisabledProps))
+
+    server.shutdown()
+  }
+
+  @Test
+  def testClusterWithoutTieredStorageFailsOnStartupIfTopicWithTieringEnabled(): Unit = {
+    val serverProps = TestUtils.createBrokerConfigs(1, zkConnect).head
+
+    val server = TestUtils.createServer(KafkaConfig.fromProps(serverProps))
+
+    val topicProps = new Properties()
+    topicProps.setProperty(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, true.toString)
+
+    TestUtils.createTopic(zkClient = server.zkClient, topic = "batman", servers = Seq(server), topicConfig = topicProps)
+
+    server.shutdown()
+
+    assertThrows(classOf[ConfigException], () => TestUtils.createServer(KafkaConfig.fromProps(serverProps)))
+  }
+
+  @Test
+  def testClusterWithoutTieredStorageFailsOnStartupIfTopicWithTieringDisabled(): Unit = {
```

Comment:
Can the test name be changed? The test doesn't fail anywhere when `remote.storage.enable=false`. Could you remove the `FailsOnStartup` from test name?